.github/workflows/mypy.yaml

```diff
@@ -7,7 +7,7 @@ on:
   pull_request:
 
 jobs:
-  lint:
+  mypy:
```

Comment:
I mentioned this elsewhere but this name change means that we will have to edit the branch protections to ensure that `mypy` is an action that prevents merging. I chose `lint` deliberately so I could treat it like the other checks. I'm fine with us changing it so long as we remember to add it to the branch protections.